app/models/plage_ouverture.rb

```diff
@@ -47,10 +51,21 @@ def self.for_motif_and_lieu_from_date_range(motif_name, lieu, inclusive_date_ran
     results.uniq
   end
 
+  def expired?
+    # Use .expired_cached? for performance
+    (recurrence.nil? && first_day < Date.today) ||
+      (recurrence.present? && recurrence.to_hash[:until].present? && recurrence.to_hash[:until].to_date < Date.today)
+  end
+
   def available_motifs
     Motif.available_motifs_for_organisation_and_agent(organisation, agent)
   end
 
+  def verify_plage_ouverture_expire_date
```

Comment:
this name suggests it only reads, please use something that suggests it'll write like `recompute` or `refresh`